src/dlangui/graphics/glsupport.d

```diff
@@ -1678,95 +1442,234 @@ GLenum primitiveTypeToGL(PrimitiveType type) {
 }
 
 
-/// OpenGL batch buffer - to draw several triangles in single OpenGL call
-class OpenGLBatch {
-    private Tex2D _currentTexture;
-    private int _currentTextureDx;
-    private int _currentTextureDy;
-    private bool _currentTextureLinear;
-    private Rect[] _srcRects;
-    private Rect[] _dstRects;
-    private uint[] _colors;
-    /// clear buffers
-    void reset() {
-        _colors.length = 0;
-        _colors.assumeSafeAppend();
-        _srcRects.length = 0;
-        _srcRects.assumeSafeAppend();
-        _dstRects.length = 0;
-        _dstRects.assumeSafeAppend();
-        _currentTexture = null;
-        _currentTextureDx = 0;
-        _currentTextureDy = 0;
-        _currentTextureLinear = false;
-    }
-    /// draw buffered items
+
+/// OpenGL GUI rendering queue. It collects gui draw calls, fills a big buffer for vertex data and draws everything
+private final class OpenGLQueue {
+
+    OpenGLBatch[] batches;
+
+    /// OpenGL batch structure - to draw several triangles in single OpenGL call
+    private struct OpenGLBatch {
```

Comment:
Nested non-static aggregates contain hidden pointer to outer class/struct and this increase size of the aggregate. I just don't check if this pointer is really used in code.